
# I was trying to make it so that whenever my python program crashes it restarts

this is my code
import os
from subprocess import Popen
script=Popen('Voice Assistant Liam.py')
while 1:
    script.communicate()
    if script.returncode==0:
        os.system('cmd/k"Voice Assistant Liam.py"')


and when i try to run it i get this error
Traceback (most recent call last):
  File "C:/Users/Gavin Nelson/AppData/Roaming/Microsoft/Windows/Start Menu/Programs/Startup/Restart Liam.py", line 3, in <module>
    script=Popen('Voice Assistant Liam.py')
  File "C:\Users\Gavin Nelson\AppData\Local\Programs\Python\Python310\lib\subprocess.py", line 969, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "C:\Users\Gavin Nelson\AppData\Local\Programs\Python\Python310\lib\subprocess.py", line 1438, in _execute_child
    hp, ht, pid, tid = _winapi.CreateProcess(executable, args,
OSError: [WinError 193] %1 is not a valid Win32 application


can someone help me with this

        